pyuvdata/uvdata/uvfits.py

```diff
@@ -447,24 +446,19 @@ def read_uvfits(
                 self.channel_width = np.transpose(
                     np.tile(abs(fq_hdu.data["CH WIDTH"]), (uvfits_nchan, 1))
                 ).flatten()
-                self.freq_array = np.reshape(
-                    np.transpose(
-                        (
-                            ref_freq
-                            + fq_hdu.data["IF FREQ"]
-                            + np.outer(np.arange(uvfits_nchan), fq_hdu.data["CH WIDTH"])
-                        )
-                    ),
-                    (1, -1),
-                )
+                self.freq_array = np.transpose(
+                    (
+                        ref_freq
+                        + fq_hdu.data["IF FREQ"]
+                        + np.outer(np.arange(uvfits_nchan), fq_hdu.data["CH WIDTH"])
+                    )
+                ).flatten()
             else:
+                # If there's only one window, then the UVFITS file may not have an
+                # FQ table, in which case pull the info from the main table
```

Comment:
Should we read the FQ table if it exists? Looks like we're not even checking for it if there's only 1 spw?

Reply:
Fair point -- I've changed the logic in 3d165149eedf450d84471897c29e43afbc2767cd to look for the table in question.